Partition (WBP)/Functions/VI0039C.VTI0039.fun.yml

```diff
@@ -58,14 +66,705 @@
+    Endroutine
+    
+    *
+    * This routine returns boolean value true if for the given key, the process is able to update the record from table VTLI0039C
+    *
+    Mthroutine Name(Record_From_TableC_Is_Update_Successful)
+    Define_Map For(*BOTH) Class(#BIFA002A) Name(#this_rc)
+    Define_Map For(*RESULT) Class(#BIFB001A) Name(#this_is_ok)
+    
+    #this_is_ok := false
+    If (#Com_Owner.Fetch_Record_From_TableC( #this_rc ) *And (#BIFN0700A = #BIFN0700A_New))
+    #Com_Owner.mthAdd_SubMsg( OK ('-  Record of table VTLI0039C with key #BIFP0700A ' + #BIFP0700A.AsString + ' has been updated as #BIFN0700A ' + #BIFN0700A.AsString) ('') )
+    #this_is_ok := true
+    Endif
+    #BIFN0700A := #BIFN0700A_New
+    Endroutine
+    
+    *
+    * This routine is responsible to validate the tests for Mixed Mode Commitment control for tables VTLI0039C and VTLI0039D
+    *
     Mthroutine Name(mthMixedModeCommitmentControl)
+    Define_Map For(*BOTH) Class(#BIFA002A) Name(#this_rc)
+    
     #Com_Owner.mthAdd_SubMsg( OK ('- Mixed Mode Commitment Control tests for DataBase type ' + #LF_DB_Type) '' )
     
     If (#LF_DB_Type *EQ SQLANYWHERE)
     #Com_Owner.mthAdd_SubMsg( OK ('- Skipping tests') '' )
     Else
     #Com_Owner.mthAdd_SubMsg( OK ('- Running tests') '' )
-    * To be implemented
-    #Com_Owner.mthAdd_SubMsg( OK ('- Tests successful') '' )
+    
+    #Com_Owner.mthClearDataFromTables( #LF_rc )
+    
+    * INSERT C
+    * ROLLBACK
+    * INSERT D
+    * FETCH C - check row does not exist
+    * FETCH D - check row exists
+    * DELETE D
+    
+    #BIFP0700A := 1
+    #Com_Owner.mthAdd_SubMsg( OK ('- Initialized tables') ('') )
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Adding a Record to Table VTLI0039C') ('') )
+    If (#Com_Owner.Add_Record_To_TableC( #LF_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Record exists') ('') )
+    Endif
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Rolling back the changes') ('') )
+    Rollback
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Adding a Record to Table VTLI0039D') ('') )
+    If (#Com_Owner.Add_Record_To_TableD( #LF_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Record exists') ('') )
+    Endif
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Verifying data from table VTLI0039C. It must not have the record with Key as #BIFP0700A ' + #BIFP0700A.AsString) ('') )
+    If (#Com_Owner.Fetch_Record_From_TableC_Has_Failed( #this_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK. Record not found') ('') )
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Record exists') ('') )
+    Endif
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('- Verifying data from Table VTLI0039D. It must have the record with Key as #BIFP0700A ' + #BIFP0700A.AsString) ('') )
+    If (#Com_Owner.Fetch_Record_From_TableD_Is_Successful( #this_rc ))
+    #Com_Owner.mthAdd_SubMsg( OK ('- OK') '' )
```

Comment:
You don't need OK twice and use a description that matches the other messages.
#Com_Owner.mthAdd_SubMsg( OK ('- Record exists') '' )

Same for line 491 above and other messages below.